### DISCUSSION TITLE
Fair scheduling

### DISCUSSION THREAD

**Discussion by haines** [riverqueue/river#628] [Ideas]
This is somewhat related to https://github.com/riverqueue/river/issues/551 and https://github.com/riverqueue/river/discussions/626.

We have a multi-tenanted platform where we don't want jobs being run for a very busy tenant to starve other tenants of workers, but at the same time we want to make use of all the worker capacity available.

Is there a sensible way to achieve this with River as-is, or is that a feature that could be considered in future?

**Comment by bgentry** [comment]
Hi @haines, we've made a big step forward in addressing some of these use cases with the new [concurrency limits](https://riverqueue.com/docs/pro/concurrency-limits) feature in River Pro. It would let you limit any given customer's concurrency on a particular queue or queue + job type, though it doesn't allow bursting to take advantage of unused worker slots.

I hope you still find it useful!

**Reply by haines** [reply]
Awesome, thanks... we'll check it out!
